src/renate/updaters/learner_components/losses.py

```diff
@@ -376,68 +276,42 @@ def _loss(
         self._iteration += 1
-        if (
-            torch.rand(1, device=self._plastic_model_update_probability.device)
-            < self._plastic_model_update_probability
-        ):
+        if torch.rand(1) < self._plastic_model_update_probability:
             self._update_model_variables(
                 self._plastic_model, model, self._plastic_model_update_weight
             )
 
-        if (
-            torch.rand(1, device=self._stable_model_update_probability.device)
-            < self._stable_model_update_probability
-        ):
+        if torch.rand(1) < self._stable_model_update_probability:
             self._update_model_variables(
                 self._stable_model, model, self._stable_model_update_weight
             )
 
-    def set_stable_model_update_weight(self, stable_model_update_weight: float) -> None:
-        self._stable_model_update_weight.data = torch.tensor(
-            stable_model_update_weight,
-            dtype=self._stable_model_update_weight.dtype,
-            device=self._stable_model_update_weight.device,
-        )
-        self._verify_attributes()
-
-    def set_plastic_model_update_weight(self, plastic_model_update_weight: float) -> None:
-        self._plastic_model_update_weight.data = torch.tensor(
-            plastic_model_update_weight,
-            dtype=self._plastic_model_update_weight.dtype,
-            device=self._plastic_model_update_weight.device,
-        )
-        self._verify_attributes()
-
-    def set_stable_model_update_probability(self, stable_model_update_probability: float) -> None:
-        self._stable_model_update_probability.data = torch.tensor(
-            stable_model_update_probability,
-            dtype=self._stable_model_update_probability.dtype,
-            device=self._stable_model_update_probability.device,
-        )
-        self._verify_attributes()
-
-    def set_plastic_model_update_probability(self, plastic_model_update_probability: float) -> None:
-        self._plastic_model_update_probability.data = torch.tensor(
-            plastic_model_update_probability,
-            dtype=self._plastic_model_update_probability.dtype,
-            device=self._plastic_model_update_probability.device,
-        )
-        self._verify_attributes()
+    def on_load_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """Load relevant information from checkpoint."""
+        super().on_load_checkpoint(checkpoint)
+        self._plastic_model = checkpoint["component-cls-plastic-model"]
+        self._stable_model = checkpoint["component-cls-stable-model"]
+        self._iteration = checkpoint["component-cls-iteration"]
+
+    def on_save_checkpoint(self, checkpoint: Dict[str, Any]) -> None:
+        """Add plastic and stable model to checkpoint."""
+        super().on_save_checkpoint(checkpoint)
+        checkpoint["component-cls-plastic-model"] = self._plastic_model
```

Comment:
We should save and load `self._plastic_model.state_dict()` instead of the model object itself.

Reply:
done